routes/api/sites.js

```diff
@@ -26,7 +28,13 @@ router.get('/:id', async (req, res) => {
 
 router.post('/', interceptors.requireAdmin, async (req, res) => {
   try {
-    const record = await models.Site.create(_.pick(req.body, ['name', 'address', 'phoneNumber', 'email', 'website']));
+    let record;
+    await models.sequelize.transaction(async (transaction) => {
+      record = await models.Site.create(_.pick(req.body, ['name', 'address', 'phoneNumber', 'email', 'website']), { transaction });
+      if (record) {
+        await record.setNutritionPartners(req.body.NutritionPartnerIds ?? [], { transaction });
```

Comment:
Here we see how the associations are set- we just pass the ids into the setNutritionParners function (which is created dynamically by sequelize when the belongsToMany() association is set up). We now wrap the create in a transaction so that the create and set occur as one atomic operation in the database.